### DISCUSSION TITLE
EZOut Filament Sensor

### DISCUSSION THREAD

**Discussion by eecue** [Jyers/Marlin#335] [Q&A] [answered]
Hello,

I have an Th3D Studio EZOut filament sensor which connects to the buzzer pin using a little breakout board:

https://www.th3dstudio.com/product/ender-3-ender-3-v2-and-ender-5-filament-sensor-kit/ 

I don't think the firmware is recognizing it, it definitely lights up when I have filament in there but everything is running fine even though I currently am not routing the filament through it.  I have the Filament Sensor checkbox checked in the interface.

Thanks in advance and thanks for all the work you've put into supporting this!

Dave

**UPDATE** - Ok so turns out that the EZOut has swapped pins, if you plug it directly into the board as is, it has the V and S pins swapped which causes the system to crash when filament runs out.  Swapping the pins fixed the issue and it now works great!

**Comment by cmtekwerx** [comment]
Not an answer, but, I'm in the same situation with the same sensor.

However, it's my understanding that the sensor is only active once a print starts. I take that to mean it looks for a state change from the sensor. If your sensor is not inline when the print starts, it'll ignore it the state changes after. 

Like you, I suspect mine isn't working quite right. My printer is currently busy, so I can't test. Have you placed filament in it, started a print and then removed the filament to see what happens? I did on mine and it made no difference.

**Reply by eecue** [reply]
Yes I did this and it kept printing.

**Comment by joeclifford** [comment, marked as answer]
Are you using the breakout board listed in that link or attaching the sensor directly to the board?

looking at the bottom of the the install guide, https://support.th3dstudio.com/hc/product-information/ezout/ezout-installation-guide/, the wires are:

- Red = VCC (3.3-5V DC)
- Black = Ground
- White = Signal

However the pins on the 4.2.x boards are (left to right):

- Signal
- Ground
- Voltage

https://github.com/MarlinFirmware/Marlin/issues/19029#issuecomment-675498743

Looks like the pins might need swapping round.

I'd advise you do this at your own risk as messing with voltage pins can fry thing.

**Reply by eecue** [reply]
The breakout board installed between the LCD connector and the main board. I could have connected directly to the board? lol oops.

**Reply by joeclifford** [reply]
Just updated my reply with more info ^

**Reply by eecue** [reply]
What's strange is that I'm using the little breakout board and the filament sensors works as it should, when I insert filament, the green light turns on.... however the software doesn't stop the print when no filament is detected.

**Reply by eecue** [reply]
Ok just realized I missed this step: 
>  Connect the 2 pin red/black cable to the “CHECK” header next to the LCD Socket. If you do not it will not work.

**Reply by eecue** [reply]
In other threads I've seen folks saying just plugging it in directly to the board worked.  I'm going to try that.

**Reply by eecue** [reply]
Will double check the pins though and potentially re-route.

**Comment by Jyers** [comment]
Any update on this?

**Reply by eecue** [reply]
I just rewired it to the correct port on the board, testing now.

**Comment by eecue** [comment]
Ok it works fine as long as it's connected to the correct header on the board, the daughterboard I had linked above is totally not needed for the 3v2 ... btw, it just stopped printing and cancelled the print (printing via Octoprint) do I need to change something to have it pause instead?

Edit: answer to my second question here: https://github.com/Jyers/Marlin/discussions/148#discussioncomment-430101

**Comment by eecue** [comment]
Btw, I did need to swap the pins as noted above.  It was crashing when I would take the filament out.  Swapping V/S seemed to fix it.

**Comment by Jer406** [comment]
So if im reading the right all you have to do is swap the pins on the connector that goes into the board? there is no need for the  expansion board?
